nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/GetTCP.java

```diff
@@ -0,0 +1,668 @@
+        _propertyDescriptors.add(CONNECTION_TIMEOUT);
+        _propertyDescriptors.add(CONNECTION_ATTEMPT_COUNT);
+        _propertyDescriptors.add(BATCH_SIZE);
+        _propertyDescriptors.add(RECEIVE_BUFFER_SIZE);
+        _propertyDescriptors.add(KEEP_ALIVE);
+
+
+        propertyDescriptors = Collections.unmodifiableList(_propertyDescriptors);
+
+        Set<Relationship> _relationships = new HashSet<>();
+        _relationships.add(REL_SUCCESS);
+        _relationships.add(REL_FAILURE);
+        relationships = Collections.unmodifiableSet(_relationships);
+    }
+
+    private Map<SocketRecveiverThread, Future> socketToFuture = new HashMap<>();
+    private ExecutorService executorService;
+    private transient ComponentLog log = getLogger();
+    private transient String originalServerAddressList;
+    private transient String backupServer;
+    private transient int batchSize;
+
+    /**
+     * Bounded queue of messages events from the socket.
+     */
+    protected final BlockingQueue<String> socketMessagesReceived = new ArrayBlockingQueue<>(256);
+
+    @Override
+    public Set<Relationship> getRelationships() {
+        return relationships;
+    }
+
+    @Override
+    public final List<PropertyDescriptor> getSupportedPropertyDescriptors() {
+        return propertyDescriptors;
+    }
+
+    @Override
+    protected PropertyDescriptor getSupportedDynamicPropertyDescriptor(final String propertyDescriptorName) {
+        return new PropertyDescriptor.Builder()
+                .required(false)
+                .name(propertyDescriptorName)
+                .addValidator(StandardValidators.NON_EMPTY_VALIDATOR)
+                .dynamic(true)
+                .expressionLanguageSupported(true)
+                .build();
+    }
+
+
+    @OnScheduled
+    public void onScheduled(final ProcessContext context) throws ProcessException {
+
+
+        final int rcvBufferSize = context.getProperty(RECEIVE_BUFFER_SIZE).asInteger();
+        final boolean keepAlive = context.getProperty(KEEP_ALIVE).asBoolean();
+        final int connectionTimeout = context.getProperty(CONNECTION_TIMEOUT).asTimePeriod(TimeUnit.SECONDS).intValue();
+        final int connectionRetryCount = context.getProperty(CONNECTION_ATTEMPT_COUNT).asInteger();
+        originalServerAddressList = context.getProperty(ENDPOINT_LIST).getValue();
+        final String[] serverAddresses = originalServerAddressList.split(",");
+        backupServer = context.getProperty(FAILOVER_ENDPOINT).getValue();
+        executorService = Executors.newFixedThreadPool(serverAddresses.length);
+        batchSize = context.getProperty(BATCH_SIZE).asInteger();
+
+        for (final Map.Entry<PropertyDescriptor, String> entry : context.getProperties().entrySet()) {
+            final PropertyDescriptor descriptor = entry.getKey();
+            if (descriptor.isDynamic()) {
+                dynamicAttributes.put(descriptor.getName(), entry.getValue());
+
+            }
+        }
```

Comment:
Since dynamic properties are not used anywhere, you might as well remove the above loop